include/otf2xx/event/metric.hpp

```diff
@@ -256,7 +300,7 @@ namespace event
             return static_cast<bool>(metric_instance_);
         }
 
-        otf2::definition::metric_instance metric_instance() const
+        weak_ref<otf2::definition::metric_instance> metric_instance() const
```

Comment:
Add setters (`void metric_instance(weak_ref<otf2::definition::metric_instance>)`) for instance and class.